src/foam/nanos/column/NestedPropertiesExpression.js

```diff
@@ -81,61 +81,118 @@ foam.CLASS({
         {
           name: 'i',
           class: 'Int'
+        },
+        {
+          name: 'expr',
+          javaType: 'foam.mlang.Expr'
         }
       ],
-      code: function (of, propName, i) {
-        if ( i === propName.length - 1 )
-          return of.getAxiomByName(propName[i]);
+      code: function (of, propName, i, expr) {
         var prop = of.getAxiomByName(propName[i]);
-        return foam.mlang.Expressions.create().DOT(prop, this.returnPropExpr(prop, this.returnDotExprForNestedProperty(prop.of, propName, ++i)));
+        if ( ! prop ) return null;
+
+        expr = this.buildPropertyExpr(prop, expr);
+
+        if ( i === propName.length - 1 )
+          return expr;
+
+        return this.returnDotExprForNestedProperty(prop.of, propName, ++i, expr);
       },
       javaCode: `
         ClassInfo ci = of;
-        PropertyInfo p = (PropertyInfo) ci.getAxiomByName(propName[i]);
-        if ( i == propName.length - 1 ) {
-          return p;
-        }
+        PropertyInfo prop = (PropertyInfo) ci.getAxiomByName(propName[i]);
+
+        if ( prop == null ) return null;
+
+        Boolean isPropAReference = isPropertyAReference(ci, prop);
+        expr = buildPropertyExpr(prop, expr, isPropAReference);
+
+        if ( i == propName.length - 1 )
+          return expr;
         
-        StringBuilder sb = new StringBuilder("find");
         try {
+          StringBuilder sb = new StringBuilder("find");
```

Comment:
can you extract the logic in try to method as well?

Reply:
fixed